src/libraries/System.Data.Common/src/System/Data/Common/SqlUDTStorage.cs

```diff
@@ -34,7 +35,11 @@ private SqlUdtStorage(DataColumn column, Type type, object nullValue)
         }
 
         // to support oracle types and other INUllable types that have static Null as field
-        internal static object GetStaticNullForUdtType(Type type) => s_typeToNull.GetOrAdd(type, t =>
+        internal static object GetStaticNullForUdtType([DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicProperties)] Type type) => s_typeToNull.GetOrAdd(type, t => GetStaticNullForUdtTypeCore(type));
+
+        [UnconditionalSuppressMessage("ReflectionAnalysis", "IL2070:UnrecognizedReflectionPattern",
+            Justification = "The only callsite is marked as unsafe. Workaround for https://github.com/mono/linker/issues/1981")]
```

Comment:
This is not true. The only callsite is not marked unsafe.

Reply:
The Type should be annotated with `PublicProperties | PublicFields`, and then this suppression can be removed.

Reply:
This appears to still be an issue.

Reply:
I noticed I did in fact miss `| PublicFields` in the only caller but other than that i do not see how to remove this suppression. If I leave it without it then this method is used in the lambda and I'm getting:

```
Trim analysis warning IL2077: System.Data.Common.SqlUdtStorage.<>c__DisplayClass6_0.<GetStaticNullForUdtType>b__0(Type): 'type' argument does not satisfy 'DynamicallyAccessedMemberTypes.PublicFields', 'DynamicallyAccessedMemberTypes.PublicProperties' in call to 'System.Data.Common.SqlUdtStorage.GetStaticNullForUdtTypeCore(Type)'. The field 'System.Type System.Data.Common.SqlUdtStorage/<>c__DisplayClass6_0::type' does not have matching annotations. The source value must declare at least the same requirements as those declared on the target location it is assigned to. [d:\src\runtime-2\src\libraries\src.proj]
```

Reply:
What you have now looks like the right approach.